go/common/consensus/consensus.go

```diff
@@ -3,12 +3,21 @@
 package consensus
 
 import (
+	"context"
+
 	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
 	"github.com/oasislabs/oasis-core/go/common/node"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
 )
 
 // Backend is an interface that a consensus backend must provide.
 type Backend interface {
+	// Initialize initializes the consensus backends
+	// XXX: as of now, this only initializes the epochtime backend, with
+	// https://github.com/oasislabs/oasis-core/issues/1879 this should
+	// initialize all backends.
+	InitializeBackends() (epochtime.Backend, error)
```

Comment:
Is there a reason why this wouldn't happen as part of the consensus backend constructor?

The only thing that you need to expose then is a way to get the various backends, e.g.:
```golang
EpochTime() epochtime.Backend
```

Reply:
Yeah I think it would be better to initialize all backends in `New` and then the provided method would be to just return the initialized backend(s).